### PR TITLE
Converts all uses of modifiers to lazy access to avoid memes in future

### LINKED ISSUE #56848: pick-up verb no longer works
156127
either shit click and click pick up or use directly verb for item to not go in hand

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

Fixes #56848
Fixes some runtimes on master to do with uses of modifiers not checking that the list exists.
Standardizes LAZYACCESS as the proper way to access the modifiers list to avoid accidents in future.

## Why It's Good For The Game

Should make working with right click modifiers more intuitive, and less error prone. Fixes the right click pickup button not working, along with maybe a few others, that's the only one I know of explicitly.

It might be worth having some of these be regular accesses, depending on how costly the check LAZYACCESS does, I think it's worth doing like this for the sake of standardization, but I'm open to changing that.

## Changelog
:cl:
fix: Fixes the Pick Up option in the shift + right click menu not working
/:cl:

### CODE REVIEW

**Comment by Jared-Fogle** [comment]
looking at this now i think "right" and friends should be defines, if that's not too hard to shove in here while you're already reimplementing everyone of them

**Comment by LemonInTheDark** [comment]
I can do that sure, might take me a bit since my computer doesn't like indexing the codebase for some reason

File: code/_onclick/ai.dm [hunk 1]
```diff
@@ -52,19 +52,19 @@
...
 
 	var/list/modifiers = params2list(params)
-	if(modifiers["shift"] && modifiers["ctrl"])
+	if(LAZYACCESS(modifiers, "shift") && LAZYACCESS(modifiers, "ctrl"))
 		CtrlShiftClickOn(A)
 		return
-	if(modifiers["shift"])
+	if(LAZYACCESS(modifiers, "shift"))
 		ShiftClickOn(A)
 		return
```
Comment (ShizCalev) [inline code/_onclick/ai.dm:60]: no sense checking for the same modified twice
```
	if(LAZYACCESS(modifiers, "shift"))
		if(LAZYACCESS(modifiers, "ctrl")
			CtrlShiftClickOn(A)
			return
		ShiftClickOn(A)
		return
```

File: code/_onclick/click.dm [hunk 2]
```diff
@@ -79,22 +79,22 @@
...
 		CtrlShiftClickOn(A)
 		return
-	if(modifiers["middle"])
+	if(LAZYACCESS(modifiers, "middle"))
 		MiddleClickOn(A, params)
 		return
-	if(modifiers["shift"])
+	if(LAZYACCESS(modifiers, "shift"))
 		ShiftClickOn(A)
 		return
```
Comment (ShizCalev) [inline code/_onclick/click.dm:93]: Reduce duplicate checks here
```
	var/list/modifiers = params2list(params)
	if(LAZYACCESS(modifiers, "shift"))
		if(LAZYACCESS(modifiers, "middle"))
			ShiftMiddleClickOn(A)
			return
		if(LAZYACCESS(modifiers, "ctrl"))
			CtrlShiftClickOn(A)
			return
		ShiftClickOn(A)
		return
	if(LAZYACCESS(modifiers, "middle"))
		MiddleClickOn(A, params)
		return
```

File: code/_onclick/cyborg.dm [hunk 3]
```diff
@@ -18,22 +18,22 @@
...
 		ShiftMiddleClickOn(A)
 		return
-	if(modifiers["middle"])
+	if(LAZYACCESS(modifiers, "middle"))
 		MiddleClickOn(A)
 		return
-	if(modifiers["shift"])
+	if(LAZYACCESS(modifiers, "shift"))
 		ShiftClickOn(A)
 		return
```
Comment (ShizCalev) [inline code/_onclick/cyborg.dm:32]: same here
```
	var/list/modifiers = params2list(params)
	if(LAZYACCESS(modifiers, "shift"))
		if(LAZYACCESS(modifiers, "ctrl"))
			CtrlShiftClickOn(A)
			return
		if(LAZYACCESS(modifiers, "middle"))
			ShiftMiddleClickOn(A)
			return
		ShiftClickOn(A)
		return
	if(LAZYACCESS(modifiers, "middle"))
		MiddleClickOn(A, params)
		return
```

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Review by Mothblocks — CHANGES_REQUESTED** [review]
linters

**Pull request merged**